spec/generators/rolify/rolify_activerecord_generator_spec.rb

```diff
@@ -239,6 +239,13 @@ class User < ActiveRecord::Base
       subject { migration_file('db/migrate/rolify_create_admin_roles.rb') }
 
       it { should be_a_migration }
+      it do
+        if Rails::VERSION::MAJOR < 5
+          should contain "< ActiveRecord::Migration"
+        else
+          should contain "< ActiveRecord::Migration[#{Rails::VERSION::MAJOR}.#{Rails::VERSION::MINOR}]"
```

Comment:
Perhaps it would be better to test the output string here? 🤔 

This test basically depends on Appraisals doing its job.

I'd maybe instead see if I can mock `Rails::VERSION::MAJOR` constant to return 4 or 5 (two tests) and then I would avoid having the exact same code in the test like in the implementation: `if Rails::VERSION::MAJOR < 5` (no test is testing this if directly)

Reply:
I'm not sure what you have in mind here. Could you elaborate?

I agree that it's not ideal to rely on Appraisals behaving appropriately. We are testing the output string though - that gets loaded as the `subject` on line 239, just outside this diff.

Reply:
What I'm saying is that, there are no two explicit tests: one for rails < 5 and one for rails >= 5.
There's one test, depending on external factor, that chooses what to test.

I think it's better if you explicitly have two tests:
when rails < 5 then test this
when rails >= 5 then test this

This test, the way it's written, it's really close to the actual code it's testing:
https://github.com/RolifyCommunity/rolify/pull/513/files/d62722bb72cbfa45ef18cede760d87255a54cec3#diff-e027c7aa91a73f0b9db8ac8dffe8c95fR86

According to rspec documentation, maybe something like this would work better:

```rb
context 'rails < 5' do
  before { stub_constant("Rails::VERSION::MAJOR", 4) }
  it { should contain "< ActiveRecord::Migration" }
end
context 'rails >= 5' do
  before { stub_constant("Rails::VERSION::MAJOR", 5) }
  it { should contain "< ActiveRecord::Migration[#{Rails::VERSION::MAJOR}.#{Rails::VERSION::MINOR}]" }
end
```

https://relishapp.com/rspec/rspec-mocks/v/2-11/docs/stubbing-constants/stub-defined-constant

I just noticed that there are other tests written in a similar way like yours. 🤷‍♂ Maybe I'm overthinking this... :)

Reply:
Yeah, I copied my approach based on other tests I saw in the project.

I do prefer your suggested technique, though I don't know how that would interact with the setup work for that test. There's only one way to find out for sure - I'll try it.

Reply:
I had to drop this back to my original approach; it looks like RSpec isn't able to override that constant (maybe I'm doing something wrong).

When I added Rails 6 final, the tests failed because the output string included `ActiveRecord::Migration[6.0]` instead of the stubbed `[5.0]`. Not to worry, we'll trust to Appraisals doing the right thing instead.